tests/cargo-kani/phantom-data/src/lib.rs

```diff
@@ -0,0 +1,17 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+//! Check that we can codegen structs with scalar and phantom data.
+
+use std::marker::PhantomData;
+
+pub struct Foo<R> {
+    x: u8,
+    _t: PhantomData<R>,
+}
+
+#[kani::proof]
+fn main() {
```

Comment:
nit: Can you please provide a more meaningful name to your check? We should stop using `main` everywhere since this is not a patter we expect others to follow.

Reply:
we also need a test with ZST that are not PhantomData

Reply:
We now have properly named tests for both ZST and `PhantomData`. It seemed a bit redundant to me since `PhantomData` is just a special case of ZST, but it doesn't hurt to keep both.